File: common/src/main/java/generations/gg/generations/core/generationscore/common/world/level/block/entities/LootClaim.java
package generations.gg.generations.core.generationscore.common.world.level.block.entities;

import java.time.Instant;
import java.util.UUID;

public record LootClaim(UUID uuid, Instant time) {
    public boolean isExpired() {
        return Instant.now().isAfter(time);
    }
}

File: common/src/main/java/generations/gg/generations/core/generationscore/common/world/level/block/entities/LootClaim.kt
package generations.gg.generations.core.generationscore.common.world.level.block.entities

import generations.gg.generations.core.generationscore.common.GenerationsCore
import java.time.Instant
import java.util.*

data class LootClaim(@JvmField val uuid: UUID, @JvmField var time: Instant) {
    val isExpired: Boolean
        get() = Instant.now().isAfter(time)

    fun refresh() {
        time = Instant.now().plus(GenerationsCore.CONFIG.lootTime)
    }
}
